client/asset/eth/eth.go

```diff
@@ -145,6 +145,7 @@ type ethFetcher interface {
 	initiate(ctx context.Context, contracts []*asset.Contract, maxFeeRate uint64, contractVer uint32) (*types.Transaction, error)
 	shutdown()
 	syncProgress() ethereum.SyncProgress
+	isRedeemable(secretHash, secret [32]byte, contractVer uint32) (bool, error)
```

Comment:
This is good for me, but I have a suspicious that `redeem` can call the contractor method of `isRedeemable` internally.